dqc_us_rules/dqc_us_0018.py

```diff
@@ -181,7 +173,9 @@ def deprecated_facts_errors(val):
     :return: No explicit return, though error messages are created
     :rtype: None
     """
-    _load_cache(val)
+    cache_exists = _load_cache(val)
+    if not cache_exists:
+        _create_cache(val)
```

Comment:
Yes.  The cache is loaded into val.usgaapDeprecations, then dumped to a file.  _load_cache takes the file's content and saves it under val.usgaapDeprecations.